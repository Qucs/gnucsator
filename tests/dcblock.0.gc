load c_include.so
load d_trln.so
load d_dcblock.so
load s_sparam.so
load mgsim/d_va_noise.so
load mgsim/d_vaflow.so
load mgsim/d_vapot_br.so

`include ../include/sources.v
`include ../include/lumped.v

verilog

module main (1,2,3);
  // TODO: ground //
  Vac #(.ampl(1)) v(1, 1a)
  Vdc #(.U(1.)) v(1a, 0)
  DCBlock #() d(1, 2)
  resistor #(.r(1)) r(2, 0)
endmodule // main

main #() m(1,2,3)

print dc v(m.d)
dc

print tran v(nodes)
tran 1

print ac vr(nodes) vp(nodes) vm(nodes)
ac 1e6 2e6 * {sqrt(2)}
