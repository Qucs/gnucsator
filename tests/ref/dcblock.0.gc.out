#           v(m.d)    
 0.         1.        
#Time       v(1)       v(2)       v(3)      
 0.         1.         0.         0.        
 0.01       1.0628     0.062789   0.        
 0.02       1.1253     0.12533    0.        
 0.036333   1.2263     0.22631    0.        
 0.12394    1.7024     0.70236    0.        
 0.21155    1.9709     0.97093    0.        
 0.34295    1.8342     0.83422    0.        
 0.47436    1.1604     0.16038    0.        
 0.60577    0.38328   -0.61672    0.        
 0.73718    0.0032614 -0.99674    0.        
 0.86859    0.26501   -0.73499    0.        
 1.         1.         0.         0.        
#Freq       vr(1)      vr(2)      vr(3)      vp(1)      vp(2)      vp(3)      vm(1)      vm(2)      vm(3)     
 1.Meg      0.99998    0.99998    0.         0.         0.         0.         0.99998    0.99998    0.        
 1.4142Meg  0.99998    0.99998    0.         0.         0.         0.         0.99998    0.99998    0.        
 2.Meg      0.99998    0.99998    0.         0.         0.         0.         0.99998    0.99998    0.        

